lib/UR/Context.pm

```diff
@@ -2603,33 +2603,15 @@ sub has_changes {
 }
 
 sub commit {
-
-    Carp::carp 'UR::Context::commit() called as a function, not a method.  Assumming commit on current context' unless @_;
-
-    my $self = shift;
-    $self = UR::Context->current() unless ref $self;
-
-    $self->__signal_change__('precommit');
-
-    unless ($self->_sync_databases) {
-        $self->__signal_observers__('sync_databases', 0);
-        $self->__signal_change__('commit',0);
-        return;
-    }
-
-    $self->__signal_observers__('sync_databases', 1);
-
-    unless ($self->_commit_databases) {
-        $self->__signal_change__('commit',0);
-        die "Application failure during commit!";
-    }
-    $self->__signal_change__('commit',1);
-
-    foreach ( $self->all_objects_loaded('UR::Object') ) {
-        delete $_->{'_change_count'};
+    my $self;
+    if (@_) {
+        $self = shift;
+        ref($self) && Carp::croak('UR::Context->commit() must be called as a class method, not an instance method');
+    } else {
+        Carp::carp 'UR::Context::commit() called as a function, not a method.  Assumming commit on current context' unless @_;
     }
-
-    return 1;
+    $self = UR::Context->current();
+    $self->commit();
```

Comment:
This change probably requires that `workflow-wrapper.pl` in `Flow` be updated to call `UR::Context::Process->commit` instead of `UR::Context->commit`. I suspect in `Workflow` and `Genome` any code that calls `Genome::Context->commit` also intends for it to commit at the process level.

Would it make sense for `UR::Context->commit` to always refer to the process like `UR::Context->rollback` does?  `UR::Context->current->commit` would still commit whatever the current transaction is.